File: verify_ui.js
const { chromium } = require('playwright');
const fs = require('fs');

(async () => {
    const browser = await chromium.launch();
    const page = await browser.newPage();
    await page.goto('http://localhost:8080/?test=true');

    // Wait for game to load
    await page.waitForSelector('#game-container');
    await page.click('#btn-new-run');

    // Open Inventory
    await page.click('#btn-inventory');
    await page.waitForSelector('#inventory-window');

    // Screenshot Inventory Centered
    await page.screenshot({ path: 'verification_inventory.png' });

    // Close Inventory
    await page.click('.modal-overlay'); // Click overlay to close (test new functionality)

    // Inject Battle and open it
    await page.evaluate(() => {
        const sceneMap = window.sceneManager.currentScene();
        const battleScene = new window.Scene_Battle(
            window.dataManager,
            window.sceneManager,
            window.windowManager,
            sceneMap.party,
            sceneMap.battleManager,
            sceneMap.windowLayer,
            sceneMap.map,
            0, 0
        );
        window.sceneManager.push(battleScene);
        battleScene.start();
    });

    await page.waitForSelector('.window-header span:has-text("Battle – Stillnight")');

    // Screenshot Battle Centered
    await page.screenshot({ path: 'verification_battle.png' });

    // Try to close battle (shake)
    const overlay = page.locator('.modal-overlay').last(); // Battle overlay
    await overlay.click();

    // Take a screenshot immediately to catch shake (might be tricky)
    await page.screenshot({ path: 'verification_battle_shake.png' });

    await browser.close();
})();

File: verify_ui.py
from playwright.sync_api import sync_playwright, expect

def verify_ui():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.goto("http://localhost:8080")

        # Wait for game container
        page.wait_for_selector("#game-container")

        # Click New Run to ensure UI is active
        page.get_by_role("button", name="New Run").click()
        page.wait_for_timeout(500) # Wait for anims

        # Screenshot 1: Main HUD (checking tile size, location art, ? button)
        page.screenshot(path="verification_hud.png")
        print("HUD screenshot taken.")

        # Click Help button
        help_btn = page.locator("#btn-help")
        help_btn.click()

        # Wait for help window
        page.wait_for_selector("#help-window")
        page.wait_for_timeout(500)

        # Screenshot 2: Help Window
        page.screenshot(path="verification_help.png")
        print("Help Window screenshot taken.")

        browser.close()

if __name__ == "__main__":
    verify_ui()
